ext/readline/readline.c

```diff
@@ -200,7 +200,7 @@ PHP_FUNCTION(readline_info)
 					if (rl_line_buffer) {
 						free(rl_line_buffer);
```

Comment:
Prior to this line, we need to `oldstr = strdup(rl_line_buffer)`.

Reply:
ok taking inspiration on this.